feature-libs/customer-ticketing/core/facade/customer-ticketing.service.ts

```diff
@@ -103,29 +107,39 @@ export class CustomerTicketingService implements CustomerTicketingFacade {
     return [GetTicketsQueryResetEvents];
   }
 
-  protected createTicketEventCommand: Command<TicketEvent, TicketEvent> =
-    this.commandService.create<TicketEvent, TicketEvent>(
-      (payload) =>
-        this.customerTicketingPreConditions().pipe(
-          switchMap(([customerId, ticketId]) =>
-            this.customerTicketingConnector
-              .createTicketEvent(customerId, ticketId, payload)
-              .pipe(
-                tap(() => {
-                  if (payload.toStatus?.id) {
-                    this.eventService.dispatch(
-                      { status: payload.toStatus?.id },
-                      TicketEventCreatedEvent
-                    );
-                  }
-                })
-              )
-          )
-        ),
-      {
-        strategy: CommandStrategy.Queue,
-      }
-    );
+  protected createTicketEventCommand: Command<
+    { ticketEvent: TicketEvent; containsAttachment?: boolean },
+    TicketEvent
+  > = this.commandService.create<
+    { ticketEvent: TicketEvent; containsAttachment?: boolean },
+    TicketEvent
+  >(
+    (payload) =>
+      this.customerTicketingPreConditions().pipe(
+        switchMap(([customerId, ticketId]) =>
+          this.customerTicketingConnector
+            .createTicketEvent(customerId, ticketId, payload.ticketEvent)
+            .pipe(
+              tap(() => {
+                if (payload.ticketEvent.toStatus?.id === STATUS.CLOSED) {
+                  this.eventService.dispatch({}, TicketClosedEvent);
+                } else if (
+                  (payload.ticketEvent.toStatus?.id === STATUS.OPEN ||
+                    payload.ticketEvent.toStatus?.id === STATUS.INPROCESS) &&
+                  !payload.containsAttachment
+                ) {
+                  this.eventService.dispatch({}, TicketReopenedEvent);
+                } else if (!payload.containsAttachment) {
+                  this.eventService.dispatch({}, NewMessageEvent);
+                }
```

Comment:
Although we should avoid negative conditions, this reads better:

```suggestion
                } else if (!payload.containsAttachment){
                   if (payload.ticketEvent.toStatus?.id === STATUS.OPEN ||
                    payload.ticketEvent.toStatus?.id === STATUS.INPROCESS) {
                        this.eventService.dispatch({}, TicketReopenedEvent);
                    } else {
                        this.eventService.dispatch({}, NewMessageEvent);
                    }
                }
```